### DISCUSSION TITLE
The current features/tools are not enough for server moderators to efficiently do their jobs.

### DISCUSSION THREAD

**Discussion by AtFreezingPoint** [FakeFishGames/Barotrauma#9476] [Ideas and suggestions]
Greetings, after almost a year of being a moderator on a high-ranking server in the game, i was able to pinpoint a few issues that the game in itself had that made the work of people hosting and operating the servers at least slightly harder. I will list a few ideas of mine that (at least, in my opinion) would help make the jobs of hosts and operators easier meanwhile improving the general quality of the services they will be able to provide.

 Starting from one, the characters that staff members are able to use on most stuff are limited for no reason, this one seems to be based on how the game wants to behave at the time for pretty much no explanation without actually further looking into the reason behind the reason on it happening. As an example, when a ban is issued, the characters "/ \ * ' ^" seem to sometimes render and be shown to the receiver of the ban and sometimes leave things like links looking like the one issuing the ban had a stroke. Another thing i want to point out on this is that there is no way for the ones responsible of handling things to embed links into stuff like bans and/or descriptions of the servers they are responsible for; meanwhile it is true that embedding links may and will eventually result in somebody abusing the system to get people to click malicious links, i myself do not think that will be much of an issue in the long-run, at least giving people the option to embed direct links to Steam would be nice to have. 
 
 The second issue i was able to pinpoint is on the staff not having enough power to effectively monitor what is going on in the round, most of the logs given by the game are too vague and not inclusive and the "used item" logs are a prime example on that whole situation; while the game rarely gives out at least a considerable amount of information on what is happening (**Player A** used item on **Player B**) the logs handed out can pretty much be considered random, most of the time the game doesn't even give out  who the item was used on (which i assume, is not supposed to be) and make moderation almost impossible as the term using an item on somebody mostly means injecting any drug to the said person and one might just cheese the entire system by using eggs to any person and calyxanide on themselves; same going with any other poison and the cure. Although the used item logs are the main culprit on this whole situation, i would never say the other types of logs that the game provides are perfect; the "attacked by" logs are nowhere near perfect, although they give out basic information like damage and the type of damage they do not detail the weapon used, the range at the time of the weapon being used, the type of ammo that was used and how much health the receiver of the attack had left after, all of which i would consider to be basic information mandatory to maximize the efficiency of the logs kept. (Even if all this is not to be shown at first glance, at least hovering over a log should show what happened in detail.) Also, the ID's just showing up as "ID Card" can be given as another example to this.

 Another issue i myself and many of my colleagues faced was the lack of documentation on most console commands, the prime example of this would be the "unbanip" command, although on the surface it is only supposed to well, unban the end user based on their ip address, it also works on SteamID's of targets and that is not documented anywhere within the game (and the official wiki as they just detail the commands using the descriptions available through the help command in-game), other examples to this issue can be seen after just breezing through the official wiki page on console commands, starting from the not-at-all documented ones; the ones that have no description at all.

 Continuing on and again, in my opinion, there should be a way to mute people on text-chat and maybe even delete their past messages somehow to prevent spam and/or any malicious activity; possibly a chat-filter in the settings to censor words or one for the server to set-up. This one should be fairly easy on the developers backs, too.

 Although i am unsure on if this one is already implemented or not, a way for server owners/hosts to opt-in to being able to auto-send analytics that would help carry the development of the game forwards to the developers would be nice.

 A way for staff members to communicate in-game without disturbance would also be nice for people that do not want to use external software for communication when on the server. Although, that would undoubtedly get abused in game-modes like Traitor.

 Generally, the player-base doesn't interfere nor care much about the ones that work dutifully to keep the individual communities free from harmful material and help the community as a whole unite, in the end encouraging, influencing the whole Barotrauma community on being a fun, welcoming place for everybody to have a place in; i am most-certain on the developers taking care of the ones responsible of uniting the community, uniting **us** together for the good of everybody by further improving what they were already able to build to this day and not letting us down, thanks all for reading and as this is supposed to be a discussion please do not forget to comment down below on what you agree on, what you do not agree on and also do not forget to express your own thoughts on anything related to the game here on GitHub, thanks again, have a nice day.

**Comment by TopDogRolly** [comment]
log alert would be awesome

**Comment by JoneKone** [comment]
Duplicate, but a valid point.
https://github.com/Regalis11/Barotrauma/discussions/8585

**Reply by AtFreezingPoint** [reply]
That guy mostly means other things by administration though, it would be pretty cool to have additional features that were made from scratch rather than improving the other already (mostly) ok ones (at least for now).
-G
